#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.20851066080816993 -0.39260319570481284 -0.8957600320686946 5.370332732514704
  position -7.53340688431059 -10.970687624329749 3.886294105518971
}
TexturedBackground {
  texture "mars"
}
TexturedBackgroundLight {
  luminosity 2
}
UnevenTerrain {
  translation 0 0 -4.44
  appearance SandyGround {
    colorOverride 0.55 0.55 0.4
    textureTransform TextureTransform {
      scale 50 50
    }
  }
}
Solid {
  translation 4.484592699562105 -5.268488667653482 -0.7957202617173638
  rotation 0.3886201340020899 0.6064340838351538 0.6936945245648226 2.373771311157481
  children [
    DEF box Shape {
      appearance DryMud {
      }
      geometry Box {
        size 1 1.5 2
      }
    }
  ]
  name "solid(2)"
  boundingObject USE box
  physics DEF phy1 Physics {
    mass 1
  }
}
Solid {
  translation 1.0157894000938816 -0.7468311484062885 -0.5074265997969637
  rotation 0.9993526963908599 -0.03187471397319774 0.016679053493664903 1.6467303310964996
  children [
    DEF box Shape {
      appearance DryMud {
      }
      geometry Box {
        size 0.8 1.7 0.9
      }
    }
  ]
  name "solid(4)"
  boundingObject USE box
  physics USE phy1
}
Solid {
  translation 1.1585192510218274 -3.2871107421326413 -1.1240136294684049
  rotation 0.8812698857958163 -0.3395742494187578 -0.3287137318718464 1.8251203193857057
  children [
    DEF box Shape {
      appearance DryMud {
      }
      geometry Box {
        size 1.5 0.9 0.9
      }
    }
  ]
  name "solid(1)"
  boundingObject USE box
  physics DEF phy1 Physics {
    mass 1
  }
}
Robot {
  translation 4.149610208636769 -1.7864768222103724 -0.9566644722331126
  rotation 0.2044369255484025 -0.11530193420622056 -0.972065330850052 0.46158079801929347
  children [
    DEF body Shape {
      appearance HammeredCopper {
      }
      geometry Box {
        size 0.7 0.5 0.2
      }
    }
    DistanceSensor {
      translation -0.35 -0.14 0
      rotation 0 0 1 -2.5
      children [
        DEF ds Shape {
          appearance ReflectiveSurface {
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "ds1"
      boundingObject USE ds
      physics Physics {
        mass 1
      }
      translationStep 0.11
      lookupTable [
        0 0 0
        0.5 1000 0
      ]
      aperture 1.5715
    }
    DistanceSensor {
      translation -0.35 0.15 0
      rotation 0 0 1 2.6763956132800115
      children [
        DEF ds2 Shape {
          appearance ReflectiveSurface {
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "ds2"
      boundingObject USE ds2
      physics Physics {
        mass 1
      }
      lookupTable [
        0 0 0
        0.5 1000 0
      ]
    }
    Solid {
      translation 0.03 0 0.19
      children [
        DEF panel Shape {
          appearance SolarCell {
          }
          geometry Box {
            size 0.6 0.5 0.01
          }
        }
        Solid {
          translation 0.25 -0.19 -0.05
          children [
            Shape {
              appearance Copper {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
          name "solar_panel_support1"
        }
        Solid {
          translation 0.25 0.19 -0.05
          children [
            Shape {
              appearance Copper {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
          name "solar_panel_support2"
        }
        Solid {
          translation -0.25 0.19 -0.05
          children [
            Shape {
              appearance Copper {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
          name "solar_panel_support3"
        }
        Solid {
          translation -0.25 -0.19 -0.05
          children [
            Shape {
              appearance Copper {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
          name "solar_panel_support4"
        }
      ]
      name "solar_panel"
    }
    Solid {
      translation -0.18 -0.24 0
      children [
        Solid {
          translation -0.00017766252744422228 -0.017340268664742475 0.05993637531167906
          rotation -0.9999999929357695 8.124469320562993e-05 -8.676267159571073e-05 1.5708626067521982
          children [
            DEF cylinder1 Shape {
              appearance DamascusSteel {
              }
              geometry Cylinder {
                height 0.09
                radius 0.02
              }
            }
            Solid {
              translation 0 0 -0.05
              children [
                DEF sphere1 Shape {
                  appearance BrushedSteel {
                  }
                  geometry Sphere {
                    radius 0.02
                  }
                }
                Solid {
                  translation 0 0.1 1.79608e-07
                  rotation -1 0 0 -1.5707953071795862
                  children [
                    DEF cylinder2 Shape {
                      appearance BrushedSteel {
                      }
                      geometry Cylinder {
                        height 0.2
                        radius 0.02
                      }
                    }
                    Solid {
                      translation 0 0 -0.11
                      children [
                        DEF sphere1 Shape {
                          appearance BrushedSteel {
                          }
                          geometry Sphere {
                            radius 0.02
                          }
                        }
                        Solid {
                          translation 0 -0.03 -9.99999e-08
                          rotation 1 0 0 1.5708
                          children [
                            DEF cylinder3 Shape {
                              appearance BrushedSteel {
                              }
                              geometry Cylinder {
                                height 0.06
                                radius 0.02
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                position 383.99754518516636
                                axis 0 0 1
                                anchor -9.51224e-05 0.00113507 0.0800056
                              }
                              device [
                                RotationalMotor {
                                  name "wheel1_left"
                                }
                              ]
                              endPoint Solid {
                                translation 0.001580723124789683 0.0010284046898156852 0.07994157767599397
                                rotation -5.869801145845215e-06 1.4992034106635618e-05 -0.9999999998703922 5.548064432512152
                                children [
                                  DEF wheel Shape {
                                    appearance DamascusSteel {
                                    }
                                    geometry Cylinder {
                                      height 0.15
                                      radius 0.15
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  mass 1
                                }
                                linearVelocity -2.3679444350083816e-05 -3.230138384621584e-05 -4.329756413458139e-06
                                angularVelocity -1.5235577337783492e-06 -3.678216595766037e-06 5.215741868236395e-06
                              }
                            }
                          ]
                          boundingObject USE cylinder3
                          physics DEF phy1 Physics {
                            mass 1
                          }
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE sphere1
                      physics USE phy1
                    }
                  ]
                  boundingObject USE cylinder2
                  physics USE phy1
                }
              ]
              boundingObject USE sphere1
              physics USE phy1
            }
          ]
          boundingObject USE cylinder1
          physics USE phy1
        }
      ]
      name "wheel1_left_system"
      physics DEF phy1 Physics {
        mass 1
      }
    }
    Solid {
      translation 0.18 -0.24 0
      children [
        Solid {
          translation -0.0001757815149209252 -0.017333744024692788 0.0600545153188472
          rotation -0.9999999930027248 7.374603208071159e-05 -9.249904417794078e-05 1.5708786387286822
          children [
            DEF cylinder1 Shape {
              appearance DamascusSteel {
              }
              geometry Cylinder {
                height 0.09
                radius 0.02
              }
            }
            Solid {
              translation 0 0 -0.05
              children [
                DEF sphere1 Shape {
                  appearance BrushedSteel {
                  }
                  geometry Sphere {
                    radius 0.02
                  }
                }
                Solid {
                  translation 0 0.1 1.79608e-07
                  rotation -1 0 0 -1.5707953071795862
                  children [
                    DEF cylinder2 Shape {
                      appearance BrushedSteel {
                      }
                      geometry Cylinder {
                        height 0.2
                        radius 0.02
                      }
                    }
                    Solid {
                      translation 0 0 -0.11
                      children [
                        DEF sphere1 Shape {
                          appearance BrushedSteel {
                          }
                          geometry Sphere {
                            radius 0.02
                          }
                        }
                        Solid {
                          translation 0 -0.03 -9.99999e-08
                          rotation 1 0 0 1.5708
                          children [
                            DEF cylinder3 Shape {
                              appearance BrushedSteel {
                              }
                              geometry Cylinder {
                                height 0.06
                                radius 0.02
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                position 77.35711276827423
                                axis 0 0 1
                                anchor 0.000341778 0.000558794 0.0800055
                              }
                              device [
                                RotationalMotor {
                                  name "wheel2_left"
                                }
                              ]
                              endPoint Solid {
                                translation 0.0010766420209634869 0.000455229820118245 0.07993867002042876
                                rotation 9.903428211306545e-07 1.5775626270937922e-06 -0.9999999999982653 4.313542066853929
                                children [
                                  DEF wheel Shape {
                                    appearance DamascusSteel {
                                    }
                                    geometry Cylinder {
                                      height 0.15
                                      radius 0.15
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  mass 1
                                }
                                linearVelocity -2.3003119702742056e-05 -3.0016048967152743e-05 -4.112056097900103e-06
                                angularVelocity -1.1166390567599421e-06 -2.9535395926941865e-06 5.303135616917082e-06
                              }
                            }
                          ]
                          boundingObject USE cylinder3
                          physics DEF phy1 Physics {
                            mass 1
                          }
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE sphere1
                      physics USE phy1
                    }
                  ]
                  boundingObject USE cylinder2
                  physics USE phy1
                }
              ]
              boundingObject USE sphere1
              physics USE phy1
            }
          ]
          boundingObject USE cylinder1
          physics USE phy1
        }
      ]
      name "wheel2_left_system"
      physics USE phy1
    }
    Solid {
      translation -0.18 0.24 0
      children [
        Solid {
          translation -0.000177663 0.0173403 0.0599364
          rotation -0.9999999929357695 8.124469320562993e-05 -8.676267159571073e-05 1.5708626067521982
          children [
            DEF cylinder1 Shape {
              appearance DamascusSteel {
              }
              geometry Cylinder {
                height 0.09
                radius 0.02
              }
            }
            Solid {
              translation 0 0 0.05
              children [
                DEF sphere1 Shape {
                  appearance BrushedSteel {
                  }
                  geometry Sphere {
                    radius 0.02
                  }
                }
                Solid {
                  translation 0 0.1 1.79608e-07
                  rotation -1 0 0 -1.5707953071795862
                  children [
                    DEF cylinder2 Shape {
                      appearance BrushedSteel {
                      }
                      geometry Cylinder {
                        height 0.2
                        radius 0.02
                      }
                    }
                    Solid {
                      translation 0 0 -0.11
                      children [
                        DEF sphere1 Shape {
                          appearance BrushedSteel {
                          }
                          geometry Sphere {
                            radius 0.02
                          }
                        }
                        Solid {
                          translation 0 0.03 -9.99999e-08
                          rotation 1 0 0 1.5708
                          children [
                            DEF cylinder3 Shape {
                              appearance BrushedSteel {
                              }
                              geometry Cylinder {
                                height 0.06
                                radius 0.02
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                position 398.74322373735123
                                axis 0 0 1
                                anchor 0.00237208 0.000614304 -0.0799754
                              }
                              device [
                                RotationalMotor {
                                  name "wheel1_right"
                                }
                              ]
                              endPoint Solid {
                                translation 0.004092824134133997 0.0011854476007205047 -0.07999019576394151
                                rotation 0.7970954420866496 0.6038533399771965 -1.7529909535236365e-06 3.1416053224850993
                                children [
                                  DEF wheel Shape {
                                    appearance DamascusSteel {
                                    }
                                    geometry Cylinder {
                                      height 0.15
                                      radius 0.15
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  mass 1
                                }
                                linearVelocity -2.7898956339633813e-05 -3.016840658202265e-05 -4.108632684336849e-06
                                angularVelocity -2.0376685986337753e-06 -4.785203003706836e-06 5.0901748018105965e-06
                              }
                            }
                          ]
                          boundingObject USE cylinder3
                          physics DEF phy1 Physics {
                            mass 1
                          }
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE sphere1
                      physics USE phy1
                    }
                  ]
                  boundingObject USE cylinder2
                  physics USE phy1
                }
              ]
              boundingObject USE sphere1
              physics USE phy1
            }
          ]
          boundingObject USE cylinder1
          physics USE phy1
        }
      ]
      name "wheel1_right_system"
      physics DEF phy1 Physics {
        mass 1
      }
    }
    Solid {
      translation 0.18 0.24 0
      children [
        Solid {
          translation -0.000177663 0.0173403 0.0599364
          rotation -0.9999999929357695 8.124469320562993e-05 -8.676267159571073e-05 1.5708626067521982
          children [
            DEF cylinder1 Shape {
              appearance DamascusSteel {
              }
              geometry Cylinder {
                height 0.09
                radius 0.02
              }
            }
            Solid {
              translation 0 0 0.05
              children [
                DEF sphere1 Shape {
                  appearance BrushedSteel {
                  }
                  geometry Sphere {
                    radius 0.02
                  }
                }
                Solid {
                  translation 0 0.1 1.79608e-07
                  rotation -1 0 0 -1.5707953071795862
                  children [
                    DEF cylinder2 Shape {
                      appearance BrushedSteel {
                      }
                      geometry Cylinder {
                        height 0.2
                        radius 0.02
                      }
                    }
                    Solid {
                      translation 0 0 -0.11
                      children [
                        DEF sphere1 Shape {
                          appearance BrushedSteel {
                          }
                          geometry Sphere {
                            radius 0.02
                          }
                        }
                        Solid {
                          translation 0 0.03 -9.99999e-08
                          rotation 1 0 0 1.5708
                          children [
                            DEF cylinder3 Shape {
                              appearance BrushedSteel {
                              }
                              geometry Cylinder {
                                height 0.06
                                radius 0.02
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                position 400.332336404853
                                axis 0 0 1
                                anchor 0.00237208 0.000614304 -0.0799754
                              }
                              device [
                                RotationalMotor {
                                  name "wheel2_right"
                                }
                              ]
                              endPoint Solid {
                                translation 0.0020694618974257213 0.0022587834275150796 -0.07998820342901869
                                rotation -0.12756465312605605 -0.9918302572752804 5.106906749339679e-06 3.1415879869118437
                                children [
                                  DEF wheel Shape {
                                    appearance DamascusSteel {
                                    }
                                    geometry Cylinder {
                                      height 0.15
                                      radius 0.15
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  mass 1
                                }
                                linearVelocity -2.7024394444917576e-05 -2.8209229732788508e-05 -3.9168803575573e-06
                                angularVelocity -1.4236193727161505e-06 -3.559895221782179e-06 5.232401269901584e-06
                              }
                            }
                          ]
                          boundingObject USE cylinder3
                          physics DEF phy1 Physics {
                            mass 1
                          }
                        }
                      ]
                      name "solid(1)"
                      boundingObject USE sphere1
                      physics USE phy1
                    }
                  ]
                  boundingObject USE cylinder2
                  physics USE phy1
                }
              ]
              boundingObject USE sphere1
              physics USE phy1
            }
          ]
          boundingObject USE cylinder1
          physics USE phy1
        }
      ]
      name "wheel2_right_system"
      physics DEF phy1 Physics {
        mass 1
      }
    }
  ]
  boundingObject USE body
  physics DEF phy Physics {
    mass 1
  }
  controller "robo_controller"
  linearVelocity -2.5485746255098034e-05 -3.0359705263092733e-05 -4.140234127151888e-06
  angularVelocity -1.051808472837221e-08 -5.897824346916901e-07 5.571833669936807e-06
}
Solid {
  translation -16.983710383040073 -17.945534035497374 -1.3277449564510202
  rotation 0.35623300430093807 0.6892701600175885 0.6308761313887661 2.5450683891449475
  children [
    DEF box Shape {
      appearance DryMud {
      }
      geometry Box {
        size 2.5 2 0.4
      }
    }
  ]
  boundingObject USE box
  physics USE phy1
}
